src/core_plugins/kibana/public/management/sections/indices/edit_index_pattern/source_filters_table/components/add_filter/add_filter.js

```diff
@@ -0,0 +1,52 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFieldText,
+  EuiButton,
+} from '@elastic/eui';
+
+export class AddFilter extends Component {
+  static propTypes = {
+    onAddFilter: PropTypes.func.isRequired,
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      filter: '',
+    };
+  }
+
+  onAddFilter = () => {
+    this.props.onAddFilter(this.state.filter);
+    this.setState({ filter: '' });
+  }
+
+  render() {
+    const { filter } = this.state;
+
+    return (
+      <EuiFlexGroup>
+        <EuiFlexItem grow={10}>
+          <EuiFieldText
+            fullWidth
+            value={filter}
+            onChange={e => this.setState({ filter: e.target.value })}
+            placeholder="source filter, accepts wildcards (e.g., `user*` to filter fields starting with 'user')"
+          />
+        </EuiFlexItem>
+        <EuiFlexItem>
+          <EuiButton
+            isDisabled={filter.length === 0}
```

Comment:
How about `filter.trim().length ===` to account for when an empty string is entered?